src/main/daml/Daml/Finance/Claims/Lifecycle/Rule.daml

```diff
@@ -153,35 +132,59 @@ template Rule
 
 -- | HIDE
 -- Check if an instrument for the new key already exists, otherwise create it.
-tryCreateNewInstrument : Party -> Time -> InstrumentKey -> InstrumentKey -> Update ()
-tryCreateNewInstrument actor eventTime oldKey newKey =
-  do
-    existingRefCidOpt <- lookupByKey @BaseInstrument.R newKey
-    case existingRefCidOpt of
-      Some _ -> pure ()
-      None -> do
-        newInstrumentCid <- BaseInstrument.exerciseInterfaceByKey @DynamicInstrument.I oldKey
-          actor DynamicInstrument.CreateNewVersion with
-            lastEventTimestamp = eventTime; version = newKey.version; prevElections = []
-        pure ()
-
--- | HIDE
--- Check if an instrument for the new key already exists, otherwise create it.
-tryCreateNewElectionInstrument : Party -> [EventData] -> Time -> Bool -> Text -> InstrumentKey ->
+tryCreateNewInstrument : Party -> [EventData] -> Time -> Optional (Bool,Text) -> InstrumentKey ->
   InstrumentKey -> Update ()
-tryCreateNewElectionInstrument actor prevElections eventTime electorIsOwner claimTag oldKey
+tryCreateNewInstrument actor prevElections eventTime electionData oldKey
   newKey = do
     existingRefCidOpt <- lookupByKey @BaseInstrument.R newKey
     case existingRefCidOpt of
       Some _ -> pure ()
       None -> do
-        let
-          newElection = EventData with
-            t = eventTime
-            election = Some (electorIsOwner, claimTag)
         instrumentCid <- BaseInstrument.exerciseInterfaceByKey @DynamicInstrument.I oldKey
           actor DynamicInstrument.CreateNewVersion with
             lastEventTimestamp = eventTime
-            prevElections = prevElections ++ [newElection]
+            prevElections = case electionData of
+              None -> prevElections
+              Some (electorIsOwner, claimTag) -> prevElections ++ [newElection]
+                where
+                  newElection = EventData with
+                    t = eventTime
+                    election = Some (electorIsOwner, claimTag)
             version = newKey.version
         pure ()
+
+-- | HIDE
+-- Fast-forward to lastEventTimestamp and lifecycle as of the current event time.
+fastForwardAndLifecycle : InstrumentKey -> [ContractId NumericObservable.I] -> Time -> Party ->
+  Update ([TaggedClaim], [Pending], [TaggedClaim], Claim.I, DynamicInstrument.View)
+fastForwardAndLifecycle instrument observableCids eventTime lifecycler = do
+
+  -- BOND_PROCESS_CLOCK_UPDATE_INITAL_CLAIMS_BEGIN
+  claimInstrument <- fetchInterfaceByKey @BaseInstrument.R instrument
+  -- BOND_PROCESS_CLOCK_UPDATE_INITAL_CLAIMS_END
+  -- BOND_PROCESS_CLOCK_UPDATE_LIFECYCLE_FASTFORWARD_BEGIN
+  -- Recover claims tree as of the lastEventTimestamp. For a bond, this just requires
+  -- lifecycling as of the lastEventTimestamp.
+  nv <- BaseInstrument.exerciseInterfaceByKey @DynamicInstrument.I instrument lifecycler
```

Comment:
what does `nv` stand for? I can't recall this so maybe a more descriptive name makes sense